lib/wasmer-types/src/types.rs

```diff
@@ -242,8 +242,8 @@ impl FunctionType {
         Returns: Into<Vec<Type>>,
     {
         Self {
-            params: params.into(),
-            results: returns.into(),
+            params: params.into().into_boxed_slice(),
+            results: returns.into().into_boxed_slice(),
```

Comment:
Rust can infer the target type of `returns.into()` to the temporary `Vec<Type>`? Impressive.

Reply:
In this case, `params` and `returns` and generic and the _only_ thing we know about these types is that they implement `Into<Vec<Type>>`, but I agree that it is nice!